### ISSUE TITLE
BMEcat version 5.0.1 support

### ISSUE THREAD

**Issue by from2day** [warappa/OpenTransSharp#9]
Hi,

it's a very impressive package that helps to build the needed files. 
However the BMEcat ETIM version has been improved and it's on [5.0.1](https://www.etim-international.com/wp-content/uploads/2021/09/bmecat_etim_501.xsd).

Is there a plan to update the models or is there a way to adjust that in current implementation but I'm missing the documentation maybe.

E.g.: New transaction mode:  T_NEW_PRODUCTDATA, etc.

Thanks.

**Comment by warappa** [comment]
Hi @from2day,
I never worked with ETIM, so what I say below may not fully work.

As far as I see, ETIM is mostly a user defined extension, so it could be handled by the `IncludeUdxTypes` configuration extension-point and providing your custom data classes with XML annotations (`XmlRoot`, `XmlElement`,...).

With `XsdUris` it should be possible to provide a custom XSD to override some portion of the BMEcat XSDs like I did with the `CustomData.xsd` file in `BMEcatSharp.Samples.AspNetCore`. This way you should be able to make the value `T_NEW_PRODUCTDATA` a valid transaction mode.

I would be interested to hear back from you if this actually works to support ETIM.

**Comment by warappa** [comment]
I just noticed you also would need to replace the BMEcat XSD with the `bmecat_etim_50.xsd` which is currently not supported by any extension-point in BMEcatSharp.

Also all BMEcat XML attributes (in C#, like `BMEXmlElementAttribute`) are hardcoded to the official XML namespace `http://www.bmecat.org/bmecat/2005`.  
ETIM operates in another XML namespace (`https://www.etim-international.com/bmecat/50`) and mimics BMEcat elements but without using the official BMEcat namespace. This is something the XML parser is really picky about.

So unfortunately, I think this is a showstopper 😕

**Comment by from2day** [comment]
Hi, 

thanks for reply.

The ETIM BMEcat file is a bit more loose in that sense, yes. Digging more deeper into the specification of it the first impression was false - I had to treat it as a new document. In the end that led to drop the usage of this package.

But in general the XML parser does the validation part right when the right `.xsd` is given and this package pushed me to the right direction. Thanks.
I've had to model the C# classes according the ETIM BMEcat specification from scratch.

The main takeaway: BMECat != ETIM BMECat :)
